osu.Game.Tests/Visual/UserInterface/TestSceneFirstRunSetupOverlay.cs

```diff
@@ -0,0 +1,192 @@
+                performer.Reset();
+                notificationOverlay.Reset();
+
+                performer.Setup(g => g.PerformFromScreen(It.IsAny<Action<IScreen>>(), It.IsAny<IEnumerable<Type>>()))
+                         .Callback((Action<IScreen> action, IEnumerable<Type> types) => action(null));
+
+                notificationOverlay.Setup(n => n.Post(It.IsAny<Notification>()))
+                                   .Callback((Notification n) => lastNotification = n);
+            });
+
+            AddStep("add overlay", () =>
+            {
+                Child = overlay = new FirstRunSetupOverlay
+                {
+                    State = { Value = Visibility.Visible }
+                };
+            });
+        }
+
+        [Test]
+        public void TestDoesntOpenOnSecondRun()
+        {
+            AddStep("set first run", () => LocalConfig.SetValue(OsuSetting.ShowFirstRunSetup, true));
+
+            AddUntilStep("step through", () =>
+            {
+                if (overlay.CurrentScreen?.IsLoaded != false) overlay.NextButton.TriggerClick();
+                return overlay.State.Value == Visibility.Hidden;
+            });
+
+            AddAssert("first run false", () => !LocalConfig.Get<bool>(OsuSetting.ShowFirstRunSetup));
+
+            AddStep("add overlay", () =>
+            {
+                Child = overlay = new FirstRunSetupOverlay();
+            });
+
+            AddWaitStep("wait some", 5);
+
+            AddAssert("overlay didn't show", () => overlay.State.Value == Visibility.Hidden);
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestOverlayRunsToFinish(bool keyboard)
+        {
+            AddUntilStep("step through", () =>
+            {
+                if (overlay.CurrentScreen?.IsLoaded != false)
+                {
+                    if (keyboard)
+                        InputManager.Key(Key.Enter);
+                    else
+                        overlay.NextButton.TriggerClick();
+                }
+
+                return overlay.State.Value == Visibility.Hidden;
+            });
+
+            AddUntilStep("wait for screens removed", () => !overlay.ChildrenOfType<Screen>().Any());
+
+            AddStep("no notifications", () => notificationOverlay.VerifyNoOtherCalls());
+
+            AddStep("display again on demand", () => overlay.Show());
+
+            AddUntilStep("back at start", () => overlay.CurrentScreen is ScreenWelcome);
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestBackButton(bool keyboard)
+        {
+            AddAssert("back button disabled", () => !overlay.BackButton.Enabled.Value);
+
+            AddUntilStep("step to last", () =>
+            {
+                var nextButton = overlay.NextButton;
+
+                if (overlay.CurrentScreen?.IsLoaded != false)
+                    nextButton.TriggerClick();
+
+                return nextButton.Text.ToString() == "Finish";
```

Comment:
Could just do

```suggestion
                return nextButton.Text == CommonStrings.Finish;
```

Theoretically safer by not relying on ambient test culture.